### ISSUE TITLE
Heredoc delimiter containing "END" with a space is disallowed

### ISSUE THREAD

**Issue by pxeger** [rubocop/rubocop#10361]
I prefer to put spaces in my heredoc delimiters, because they're more readable than WORDSLIKETHIS. For example:

```
<<'END HTML'
  <html>
    ...
  </html>
END HTML
```

However, Rubocop's `Naming/HeredocDelimiterNaming` cop detects this as not being a "meaningful" delimiter, because it seems to think the space ends the delimiter (?).

I suggest the following regex:

```
^\s*(EO[A-Z]|END)\s*$
```

This rejects only delimiters which are *only* the strings `END` or `EOF` etc.

I can do a PR.

Related: #5403

--------

## Expected behavior

Rubocop accepts heredoc delimiters of the form `END HTML`, flagging no offences.

## Actual behavior

```
For /...: Default configuration from /.../gems/rubocop-1.24.1/config/default.yml
Use parallel by default.
Running parallel inspection
Inspecting 1 file
Scanning .../heredoc_test.rb
For /tmp: Loading cache from /.../1000/rubocop_cache/...
C

Offense:

test.rb:5:1: C: Naming/HeredocDelimiterNaming: Use meaningful heredoc delimiters.
END HTML
^^^^^^^^

1 file inspected, 1 offense detected
Finished in 0.2415 seconds
```

## Steps to reproduce the problem

```ruby
<<'END HTML'
  <html>
    ...
  </html>
END HTML
```

```bash
$ rubocop --debug test.rb
```

## RuboCop version

```bash
$ rubocop -V
1.24.1 (using Parser 3.1.0.0, rubocop-ast 1.15.1, running on ruby 3.1.0 x86_64-linux)
```

**Comment by bbatsov** [comment]
Interesting. I didn't even know this was possible. I wonder how well most editors handle it, though.

**Comment by pxeger** [comment]
Ruby will accept almost literally any string as a heredoc delimiter:
~~~ruby
puts <<'\n!"£$%^&*()-=_+[];#{}:@~,./<>?|`\'
hello
\n!"£$%^&*()-=_+[];#{}:@~,./<>?|`\
# back to normal code
~~~
(heh, GitHub doesn't parse it right either)

**Comment by pxeger** [comment]
still a bug, stupid bot

**Comment by bbatsov** [comment]
Closing as stale — heredoc delimiters with spaces are extremely niche and this has been dormant for 4+ years.
